src/libraries/Microsoft.PowerFx.Interpreter/UDF/UserDefinedTexlFunction.cs

```diff
@@ -53,6 +53,21 @@ public IEnumerable<ExpressionError> Bind()
                 var errorList = new List<ExpressionError>
                 {
                     new ExpressionError()
+                    {
+                        Message = "Expression is not a ParsedExpression"
+                    }
+                };
+                return errorList;
+            }
+
+            if (!check.ReturnType._type.Kind.Equals(ReturnType.Kind))
+            {
+                var errorList = new List<ExpressionError>
+                {
+                    new ExpressionError()
+                    {
+                        Message = "Return type does not match body return type"
```

Comment:
"The stated function return type does not match the return type of the function body"